lib/nerves_hub_link/archive_manager.ex

```diff
@@ -0,0 +1,148 @@
+defmodule NervesHubLink.ArchiveManager do
+  @moduledoc """
+  GenServer responsible for brokering messages between:
+    * an external controlling process
+    * HTTP
+
+  Should be started in a supervision tree
```

Comment:
The moduledoc sounds like it was copy/pasted from somewhere else. I was thinking this would talk about getting notifications for archives and downloading them.

One thing I'm missing is how a device determines whether it wants archive. Once it has processed the archive, does the archive stick around and if there's a new one, does the old one get deleted?

Reply:
The archive sticks around currently, the UUID is what blocks the archive manager from downloading the same file over and over. If there's a new one we'll keep the old file and download the new one. I'm not sure we can stream these like the existing firmware manager. Someone mentioned we should have a callback before downloading and let the client determine when we download files and your main application can clean up the old files afterwards then maybe?